extensions/vscode/src/api/client.ts

```diff
@@ -22,6 +22,19 @@ class PublishingClientApi {
     this.client = axios.create({
       baseURL: apiBaseUrl,
     });
+    this.client.interceptors.request.use((request) => {
+      (<any>request).ts = performance.now();
```

Comment:
Rather than making these `any` we can adjust the `axios.d.ts` module type to include `ts`:

```
declare module "axios" {
  interface AxiosRequestConfig {
    ignoreCamelCase?: string[];
    ts?: DOMHighResTimeStamp;
  }
}
```

Note that I made it optional here otherwise we have to include a default in `axios.create`. We can supply that default to remove the optionality of `ts`.

This would allow us to avoid the `(<any>request)` typing and just use the new attribute directly.

Reply:
beautiful!